hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java

```diff
@@ -0,0 +1,120 @@
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests snapshot in OM HA setup.
+ */
+@Timeout(300)
+public class TestOzoneManagerHASnapshot extends TestOzoneManagerHA {
+
+  /**
+   * Test snapshotNames are unique among OM nodes when snapshotName is not
+   * passed or empty.
+   */
+  @Test
+  public void testUniqueSnapshotName() throws Exception {
+    OzoneBucket ozoneBucket = setupBucket();
+    String volumeName = ozoneBucket.getVolumeName();
+    String bucketName = ozoneBucket.getName();
+
+    createKey(ozoneBucket);
+
+    getObjectStore().createSnapshot(volumeName, bucketName, "");
+    List<OzoneManager> ozoneManagers = getCluster().getOzoneManagersList();
+    List<String> snapshotNames = new ArrayList<>();
+
+    for (OzoneManager ozoneManager : ozoneManagers) {
+      await().atMost(Duration.ofSeconds(120))
+          .until(() -> {
+            String snapshotPrefix = OM_KEY_PREFIX + volumeName +
+                OM_KEY_PREFIX + bucketName;
+            SnapshotInfo snapshotInfo = null;
+            try (TableIterator<String, ?
+                extends Table.KeyValue<String, SnapshotInfo>>
+                     iterator = ozoneManager.getMetadataManager()
+                .getSnapshotInfoTable().iterator(snapshotPrefix)) {
+              while (iterator.hasNext()) {
+                snapshotInfo = iterator.next().getValue();
+              }
+            } catch (IOException e) {
+              throw new RuntimeException(e);
+            }
+
+            if (snapshotInfo != null) {
+              snapshotNames.add(snapshotInfo.getName());
+            }
+            return snapshotInfo != null;
+          });
+    }
+
+    assertEquals(1, snapshotNames.stream().distinct().count());
+    assertNotNull(snapshotNames.get(0));
+    assertTrue(snapshotNames.get(0).startsWith("s"));
+  }
+
+  @Test
+  public void testSnapshotChainManagerRestore() throws Exception {
+    List<OzoneBucket> ozoneBuckets = new ArrayList<>();
+    List<String> volumeNames = new ArrayList<>();
+    List<String> bucketNames = new ArrayList<>();
+
+    for (int i = 0; i < 10; i++) {
+      OzoneBucket ozoneBucket = setupBucket();
+      ozoneBuckets.add(ozoneBucket);
+      volumeNames.add(ozoneBucket.getVolumeName());
+      bucketNames.add(ozoneBucket.getName());
+    }
+
+    for (int i = 0; i < 100; i++) {
+      int index = i % 10;
+      createKey(ozoneBuckets.get(index));
+      String snapshot1 = "snapshot-" + RandomStringUtils.randomNumeric(5);
+      getObjectStore().createSnapshot(volumeNames.get(index),
+          bucketNames.get(index), snapshot1);
+    }
+
+    // Restart leader OM
+    OzoneManager omLeader = getCluster().getOMLeader();
+    getCluster().shutdownOzoneManager(omLeader);
+    getCluster().restartOzoneManager(omLeader, true);
+
+    await().atMost(Duration.ofSeconds(180))
+        .until(() -> getCluster().getOMLeader() != null);
+  }
```

Comment:
hmm. Is there supposed to be snapshot chain restore/reload validation logic towards the end of this test case?

Reply:
Added assertion that OMLeader node should be non-null.

OM restart will fail if snapshotChainRestore fails as mentioned in https://issues.apache.org/jira/browse/HDDS-8530?focusedCommentId=17720063&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17720063